plugin-maven/README.md

```diff
@@ -115,6 +115,60 @@ By default, all files matching `src/main/java/**/*.java` and `src/test/java/**/*
 </configuration>
 ```
 
+<a name="format"></a>
+
+## Applying to custom sources
+
+By default, no Ant-Style include patterns are defined.  Each element under `<format>` is a step, and they will be applied in the order specified.  Every step is optional, and they will be applied in the order specified.
+
+```xml
+<configuration>
+  <format>
```

Comment:
What do you think about not having `<format>` element? Configuration could then look like:

```
<configuration>
  <endWithNewline />
  <trimTrailingWhitespace />
  <replace>
    <name>Say Hello to Mars</name>
    <search>World</search>
    <replacement>Mars</replacement>
  </replace>
  <java>
    <eclipse>...</eclipse>
  </java>
  <scala>...</scala>
</configuration>
```
?

Reply:
I think the purpose of the `<format>` element is to do specific formatting on specific kinds of files that aren't java or scala.  e.g.:

https://github.com/diffplug/spotless/blob/4d28488b9848974c20f0723905bdc2d44fb2c627/spotlessSelf.gradle#L54-L59

Reply:
Aha, got it. This explains a lot :)
Please dismiss all my previous comments about `<format>`.

Maybe then we need an ability to configure multiple formats?

Reply:
The idea of the new format section is to provide the ability to define custom formats for e.g. property and sql files. Currently only one format section is allowed, but the gradle plugin allows unlimited format sections. This way you can define a format per source set. This comes in quite handy on larger projects.

I'm planing to extend the maven-plugin to provide unlimited format sections too.